NextUp-comedy/Calc-Nextup-comedy
# Interpret watchedStream data

### In this PR
- Reshaped `watchedStreams` in a way that fits our `watches` & `content report` DB tables.

### Response 

- Input
```ts
  {
    streamId: 'b04af683-4a67-bed0-cc7e-e7922d941b70',
    contentId: '37f90fb6-d520-4ad7-ac2c-0657630e96b2',
    duration: 210,
    content: {
      id: '37f90fb6-d520-4ad7-ac2c-0657630e96b2',
      publishDate: 1621337094000,
      title: 'Huge Davies - Live at Angel Comedy',
      permalink: '/videos/hugedavisfangel1',
    },
    user: { email: [Array], purchases: [Array] },
  },
  ```
  - Output
  ```ts
    {
      contentId: '37f90fb6-d520-4ad7-ac2c-0657630e96b2',
      watchedSeconds: 30,
      watches: [
          { 
           streamId: '19c022f5-b024-cf23-135f-509105f0d383',
           duration: 30 
          }
        ]
    }
  ```
  
  ### How to test it:
  - make sure you setup your `.env` variables.
    <details>
      <summary>click to show </summary>
      
      ```
      DATABASE_URL=xxxxxx
      TEST_DB_URL=xxxxxxxxx
      JWT_SECRET=xxxxxxxxxxxxxxxxxxxxx
      APP_MAIL=xxxxxxxxxxx@gmail.com
      MAIL_PASSWORD=xxxxxxxxxxxxxxxxx
      VIEWLLIFT_EMAIL=xxxxxxxxxxxxx@gmail.com
      VIEWLLIFT_PASSWORD=xxxxxxx
      VIEWLIFT_ENDPOINT=https://xxxxxxxxx.com/
      VIEWLIFT_WATCHES_FETCH_LIMIT=1000
      NODE_ENV=development
      ```
    </details>
  - `npm i && npm run dev`
